scripts/move_extra_scans.py

```diff
@@ -0,0 +1,178 @@
+    
+    return data['SeriesNumber']
+
+
+def get_BIDS_file_with_session_num(rawdata_dir: Path,
+                                   session_number: int) -> Path:
+    # walk json files
+    list_of_session_nums = []
+
+    for root, dirs, files in os.walk(rawdata_dir):
+        if Path(root).name == 'extra':
+            continue
+        for file in files:
+            if file.endswith('.json'):
+                session_num_found = get_series_number(Path(root) / file)
+                list_of_session_nums.append(session_num_found)
+                if int(session_num_found) == int(session_number):
+                    print()
+                    return Path(root) / file
+                
+    # print(list_of_session_nums, session_number)
+    
+    
+def put_nifti_in_the_extra(rawdata_dir: Path) -> Path:
+    extra_dir = rawdata_dir / 'extra'
+    if not extra_dir.is_dir():
+        return
+    
+    for json in extra_dir.glob('*json'):
+        json_prefix = json.name.split('.json')[0]
+        
+        other_nifti_files = rawdata_dir.glob('*/*nii.gz')
+        for other_nifti_file in other_nifti_files:
+            if other_nifti_file.parent.name == 'extra':
+                continue
+
+            nifti_prefix = other_nifti_file.name.split('.nii.gz')[0]
+            if json_prefix == nifti_prefix:
+                print(f'Moving {other_nifti_file} to extra')
+                shutil.move(other_nifti_file, extra_dir / other_nifti_file.name)
+                
+        # bval
+        other_bval_files = rawdata_dir.glob('*/*bval')
+        for other_bval_file in other_bval_files:
+            if other_bval_file.parent.name == 'extra':
+                continue
+
+            bval_prefix = other_bval_file.name.split('.bval')[0]
+            if json_prefix == bval_prefix:
+                print(f'Moving {other_bval_file} to extra')
+                shutil.move(other_bval_file, extra_dir / other_bval_file.name)        
+                
+        # bvec
+        other_bval_files = rawdata_dir.glob('*/*bvec')
+        for other_bval_file in other_bval_files:
+            if other_bval_file.parent.name == 'extra':
+                continue
+
+            bval_prefix = other_bval_file.name.split('.bvec')[0]
+            if json_prefix == bval_prefix:
+                print(f'Moving {other_bval_file} to extra')
+                shutil.move(other_bval_file, extra_dir / other_bval_file.name)        
+
+
+def move_extra_scans():
+    # Setup the Google Sheets API credentials
+    scope = ["https://spreadsheets.google.com/feeds",
+            "https://www.googleapis.com/auth/spreadsheets",
+            "https://www.googleapis.com/auth/drive.file",
+            "https://www.googleapis.com/auth/drive"]
+
+    json_loc = "/data/predict1/home/nick/json_keys/gspread_official_key.json"
+    creds = ServiceAccountCredentials.from_json_keyfile_name(json_loc, scope)
+    client = gspread.authorize(creds)
+
+    # Open the Google Spreadsheet using its name
```

Comment:
Looks good. get_gs_sheet and gs_sheet_to_df from amp_scz_mri_ss kcho branch could also be useful here.

Reply:
We should plan to merge `amp_scz_mri_ss` to `qqc.amp_scz_mri` soon.